### ISSUE TITLE
java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.firestore.DocumentReference

### ISSUE THREAD

**Issue by HelenGuov** [flutter/flutter#12989]
I tried to create a field as a reference to another document in a different collection in Cloud FireStore database. I got the exception below when running the flutter app after making changes to the database on the cloud. Does anyone know what causes the problem? 

EDIT: after I upgraded Flutter, I got the error message below
```E/AndroidRuntime( 3951): java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.firestore.DocumentReference@72eedfd```

I ran ```flutter doctor``` and got:
```
[✓] Flutter (on Linux, locale en_NZ.UTF-8, channel alpha)
    • Flutter at /home/helen/flutter
    • Framework revision e8aa40eddd (4 weeks ago), 2017-10-17 15:42:40 -0700
    • Engine revision 7c4142808c
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /home/helen/Android/Sdk
    • Platform android-26, build-tools 26.0.2
    • Java binary at: /home/helen/Downloads/android-studio-ide-162.4069837-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Android Studio (version 2.3)
    • Android Studio at /home/helen/Downloads/android-studio-ide-162.4069837-linux/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2)
    • Flutter plugin version 18.4
    • Dart plugin version 172.4343.25

[✓] Connected devices
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

```

**Comment by mikhaili** [comment]
Please provide more information
On what device do u trying to run the app?

**Comment by HelenGuov** [comment]
I got the problem when running on the Android simulator: 
![image](https://user-images.githubusercontent.com/17994153/32746374-48a4d3e0-c91a-11e7-9ffa-b955aee22382.png)
Thanks for helps :)

**Comment by Hixie** [comment]
cc @collinjackson

**Comment by GeoTob** [comment]
I've seen this as well, is this being tracked in #13043 now?

**Comment by Solido** [comment]
In my case even after removing the store entries with reference field the problem persist and the flutter app can't be restarted.

`D/AndroidRuntime(16669): Shutting down VM
E/AndroidRuntime(16669): FATAL EXCEPTION: main
E/AndroidRuntime(16669): Process: fr.figaro.front, PID: 16669
E/AndroidRuntime(16669): java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.firestore.DocumentReference@c037a4c`

**Comment by Skylled** [comment]
@Solido My first guess, without seeing your code, would be that your cache is still showing the reference.

**Comment by Solido** [comment]
For sure ! question is which one ? probably the one on the android simulator.
Any idea on how to clean it ? restarting one has no effect.

**Comment by Skylled** [comment]
You'll have to do a reinstall of your app, unfortunately. Or clear its app cache in settings.

**Comment by Solido** [comment]
Building the app with Android Studio instead of the Flutter plugin did the trick to clean the cache. Thx !

**Comment by Purus** [comment]
Can someone help me on this as it's a blocker for my first ever flutter app development.

More details on https://stackoverflow.com/questions/48995255/java-lang-illegalargumentexception-unsupported-value-exception-when-accessing

**Comment by Skylled** [comment]
Can be closed, via flutter/plugins#429

**Comment by kroikie** [comment]
@HelenGuov

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/943. Any further collaboration will be done there.
